File: Coding Challenge/Exam Challenge/Exam coding Challenge/Faculty.cpp
#include "Faculty.h"
#include "Worker.h"
#include<string>
using namespace std;
using namespace pattern;
using namespace worker;
Faculty::Faculty(){

}

void Faculty::Worker(){
  cout<<"my work is to teach and help students to reach their goals. Alse, I evaluate the student's project and assignments."<<endl;
  w.worker();
}

  void Facult::nextWorker(Worker worker){
      w = worker;

  }

Faculty::~Faculty(){

}

File: Coding Challenge/Exam Challenge/worker Java Files/Faculty.java
package pattern;

import pattern.worker.Worker;

public class Faculty extends Worker{
	
	@Override
	public void work() {
		System.out.println("My Work is to teach and help students to reach their goals. "
				+ "Als, I evaluate the student's projects and assignments.");
		w.work();
	}

	@Override
	public void nextWorker(Worker worker) {
		w = worker;
	}

}
